helix-core/src/syntax.rs

```diff
@@ -689,6 +692,7 @@ impl Syntax {
                                 edit.new_end_position,
                                 point_sub(range.end_point, edit.old_end_position),
                             );
+                            modified.entry(layer_id).or_insert(false);
```

Comment:
Why not just skip the `or_insert(false)`?

Reply:
I want to only insert `false` if `true` is not already present (and if `false` is already present no update I necessary). Thats why I use `entry(..).or_insert()`. Just using `entry` does not modify the map itself. Hower `or_insert(false)` will insert false if the key is not already present

Reply:
Now with the bitset based approach its much clearer what is happening here